model-optimizer/extensions/front/output_cut.py

```diff
@@ -32,3 +32,23 @@ def run_before(self):
 
     def find_and_replace_pattern(self, graph: Graph):
         add_output_ops(graph, graph.graph['packed_outputs'], inputs=graph.graph['user_shapes'])
+
+        # For keeping tensor names information for output nodes fake outputs
+        # are added to graph. In the following code fake outputs are removed
+        # and tensor names information is moved to output->Result edge.
+        for node in graph.get_op_nodes(needs_removal=True):
+            fw_info = None
+            for idx in node.in_edges():
+                node_idx = node.in_edge(idx)['in']
+                if node_idx in node.in_nodes():
+                    in_node = node.in_node(node_idx)
+                    fw_info_value = get_edge_attribute_between_nodes(in_node, node, 'fw_tensor_debug_info')
+                    if fw_info_value:
+                        fw_info = fw_info_value
+                        break
+            graph.erase_node(node)
+
+            if fw_info and in_node is not None:
```

Comment:
`in_node` is declared inside `if` of `for` loop. It must be declared alongside `fw_info`

Reply:
Yes, it is a bug. Added in_node declaration.